core/converter/md/md.go

```diff
@@ -51,17 +91,294 @@ func (h *MD) Convert(sbType model.SmartBlockType) (result []byte) {
 	if len(h.s.Pick(h.s.RootId()).Model().ChildrenIds) == 0 {
 		return
 	}
+	switch sbType {
+	case model.SmartBlockType_STType, model.SmartBlockType_STRelation, model.SmartBlockType_STRelationOption:
+		return nil
+	}
 	buf := bytes.NewBuffer(nil)
 	in := new(renderState)
+	h.renderProperties(buf)
 	h.renderChildren(buf, in, h.s.Pick(h.s.RootId()).Model())
 	result = buf.Bytes()
 	buf.Reset()
 	return
 }
 
+func (h *MD) renderProperties(buf writer) {
```

Comment:
please refactor this method, i suggest starting from splitting to smaller methods